test/blackbox-tests/export-invalid-names.t

```diff
@@ -0,0 +1,47 @@
+Export otherwise invalid OCaml identifiers to JavaScript
+
+  $ . ./setup.sh
+  $ cat > dune-project <<EOF
+  > (lang dune 3.8)
+  > (using melange 0.1)
+  > EOF
+  $ cat > dune <<EOF
+  > (melange.emit
+  >  (compile_flags -w -32)
+  >  (emit_stdlib false)
+  >  (preprocess (pps melange.ppx))
+  >  (target js-out))
+  > EOF
+
+  $ cat > x.ml <<EOF
+  > let [@mel.as POST] post= 1
+  > let [@mel.as "GET"] get = 2
+  > let [@mel.as "put"] other = 2
+  > let [@mel.as "class"] other = 2 (* class is reserved in JS *)
+  > EOF
+
+  $ cat > x.mli <<EOF
+  > val [@mel.as "GET"] get : int
+  > val [@mel.as "class"] other : int
+  > EOF
+
+  $ dune build @melange
+
+  $ cat _build/default/js-out/x.js
+  // Generated by Melange
+  'use strict';
+  
+  
+  var get = 2;
+  
+  var other = 2;
+  
+  var GET = 2;
+  
+  var $$class = 2;
```

Comment:
Is there a way to only export one value? This might be super confusing (plus I think might make Next 13 still error out).

Reply:
It's a bit hard to do it without changing melange-compiler-libs to propagate some more information.

Right now this is exclusively a PPX transformation.

It wouldn't be hard to add another attribute, though (or some configuration to `mel.as`). something like:

```ocaml
val get [@mel.as GET {hide = true}] : int
```

Reply:
The problem with hiding the original name is that other modules won't be able to refer to it. That's the part that would need support in compiler-libs.

Reply:
What if we renamed the attribute to `mel.for_js`? In the end, that is what the feature offers, and it would probably adjust expectations that these values can't be used from other OCaml modules.

I have a PR against this branch with this change and a small refactor to move the transformation to the `structure_item` and `signature_item` functions: https://github.com/melange-re/melange/pull/718. Feel free to pick anything pickable!